bluepysnap/morph.py

```diff
@@ -55,23 +57,39 @@ def __init__(self, morph_dir, nodes):
             self._load = lru_cache(maxsize=MORPH_CACHE_SIZE)(self._load)
 
     def _is_biophysical(self, node_id):
-        return self._nodes.get(node_id, Node.MODEL_TYPE) == "biophysical"
+        return self._population.get(node_id, Node.MODEL_TYPE) == "biophysical"
 
     def get_filepath(self, node_id):
-        """Return path to SWC morphology file corresponding to `node_id`."""
-        name = self._nodes.get(node_id, Node.MORPHOLOGY)
+        """Return path to SWC morphology file corresponding to `node_id`.
+
+        Args:
+            node_id (int/CircuitNodeIds): could be a single int or a CircuitNodeIds of length 1.
+        """
+        if not isinstance(node_id, (CircuitNodeIds, np.integer, int)):
+            raise BluepySnapError("node_id must be either a single node id or a CircuitNodeIds "
+                                  "with a single value. You provided : {}.".format(node_id))
+        if isinstance(node_id, CircuitNodeIds) and len(node_id) > 1:
+            raise BluepySnapError("node_id must be either a single node id or a CircuitNodeIds "
+                                  "with a single value. Length is {}.".format(len(node_id)))
+
+        name = self._population.get(node_id, Node.MORPHOLOGY)
+        # to allow the NodeCircuitIds as node_id
+        if not isinstance(name, six.string_types):
+            name = name.squeeze()
         return os.path.join(self._morph_dir, "%s.swc" % name)
```

Comment:
Should not this return a Path ? Maybe for another PR.

Reply:
nope because python2